HPCC-2968 ConfigMgr should allow hostnames in computer configuration

rpastrana opened the pull request:
- Fixes new/new range processing to work as originally intended
- New computer process allows hostnames
- Adds new functions for IP/hostname parsing

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: f1a7df6890cdee5a85f18dc1ad666ca98dfe22be
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.395 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

rpastrana commented:
This change in the ConfigManager might impact many components including Roxy/Thor/DFU/DALI/DZESP

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: f1a7df6890cdee5a85f18dc1ad666ca98dfe22be
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.111 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

rpastrana committed:
HPCC-2968 ConfigMgr should allow hostnames in computer configuration

- Fixes new/new range processing to work as originally intended
- New computer process allows hostnames
- Adds new functions for IP/hostname parsing

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: ae9cbcfd84981eb67a798d0d835c26ec274d61fb
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.41 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
@rpastrana Who should review this?

rpastrana commented:
@wangkx please review.

wangkx commented on esp/files/scripts/configmgr/navtree.js:
```diff
@@ -4367,12 +4396,15 @@ function ipCount(){
-
-    if (ipArr[k].match("-") != null )
+	  var currentval = ipArr[k]; 
+	  var mymatch = currentval.match(ipRangePattern);
```
Remove the last 2 lines.

wangkx reviewed (commented):
Several minor comments and a question.

rpastrana commented on esp/services/WsDeploy/WsDeployService.cpp:
```diff
@@ -4722,21 +4722,58 @@ bool CWsDeployFileInfo::handleComputer(IEspContext &context, IEspHandleComputerR
+    strCheckXPath.setf("%s/%s[%s=\"%s\"][1]", XML_TAG_HARDWARE, XML_TAG_COMPUTER, XML_ATTR_NETADDRESS, netAddress);
+
+    sNode.appendf("<" XML_TAG_COMPUTER " %s=\"%s\" %s=\"%s\" %s/>",
+                  &XML_ATTR_NAME[1], getUniqueName(pEnvRoot, sName, XML_TAG_COMPUTER, XML_TAG_HARDWARE),
```
appends "_X" where X > 1 and name_X is unique

rpastrana committed:
HPCC-2968 Add host name support - code review 1

- Removes commented out lines
- Removes dead code
- Performs general cleanup

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@wangkx please review latest changes.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 0cd23fb3880ba657ecfbfd06e4e68f8f6394e24c
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.573 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

wangkx commented:
@rpastrana looks fine.

rpastrana commented:
@richardkchapman ready to be merged.

richardkchapman commented:
@rpastrana You checked the "Requires a change to the documentation" box but not either of "I have updated the documentation accordingly", or... "I have created a JIRA ticket to update the documentation." Do we need a Jira creating?

rpastrana commented:
Doc Jira subtask: https://track.hpccsystems.com/browse/HPCC-17132
@richardkchapman
